lib/nacs-seq/manager.cpp

```diff
@@ -452,12 +471,34 @@ void Manager::update_config(const YAML::Node &config)
             auto backend_node = dev_info["backend"];
             if (!backend_node)
                 throw std::runtime_error("Invalid device info: missing `backend`.");
-            m_device_info.emplace(dev_name,
-                                  DeviceInfo{backend_node.as<std::string>(),
-                                      dev_info["config"]});
+            auto it = m_device_info.find(dev_name);
+            if (it != m_device_info.end()){
+                // If DeviceInfo already present, keep it around to ensure continuity of the restart counter
+                auto old_dev_info = it->second;
```

Comment:
Also you need `auto &` here at least.

Reply:
I assume this is why the zynq backend test failed. (That test loads different configs)